base/applications/regedit/main.c

```diff
@@ -217,6 +218,29 @@ int APIENTRY wWinMain(HINSTANCE hInstance,
         default:
             break;
     }
+
+    /* Prevent multiple instances of application */
+    hMutex = CreateMutexW(NULL, FALSE, L"Regedit");
```

Comment:
Is it compatible with what MS does? (Namely, what happens on Windows when you try to open both MS regedit and ROS regedit at the same time?)

Reply:
>Is it compatible with what MS does?

Yes @HBelusca. Acording to MSDN article for `WinMain` function: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-winmain#parameters (see `hPrevInstance` parameter reference), this should be done using a Mutex:
> If you need to detect whether another instance already exists, create a uniquely named mutex using the CreateMutex function.

And then I did as described here (using Mutex method): https://stackoverflow.com/a/8799795.
So it definitely should be compatible.

>Namely, what happens on Windows when you try to open both MS regedit and ROS regedit at the same time?

Nothing suspicious, as I see. They are functioning independent from each other. Is this wrong? :thinking:

Reply:
I was not asking about how one could potentially achieve this. I was asking how MS regedit **actually** does this, in order to be able to "interact" with it (i.e. detect that if MS regedit is already opened, and you open ROS regedit at the same time, then the latter does not show up but instead re-focuses the MS regedit window).

Reply:
Ah, I understood what you mean.
As I see, MS Regedit most of all asks for its other instances by enumerating all Regedit processes. I tried to launch two identical copies of MS Regedit from the different paths, and... launching the second copy indeed redirects me to the current running instace of Regedit. So looks like the Mutex method doesn't work with MS Regedit as expected (if it does not redirect to MS Regedit after launching our one).
Also in case when I launch our Regedit first, and then MS one, it same doesn't redirect me to the instance of our one. Is it a problem of our Regedit too, or it depends only on MS Regedit in this case? :thinking: